### DISCUSSION TITLE
Extend microsoft/vscode-remote-try-* and/or github/codespaces-* repo templates?

### DISCUSSION THREAD

**Discussion by jcbhmr** [devcontainers-community/.github#6] [Ideas]
https://github.com/orgs/microsoft/repositories?q=vscode-remote-try-&type=all&language=&sort=

- vscode-remote-try-sqlserver
- vscode-remote-try-cpp
- vscode-remote-try-dotnet
- vscode-remote-try-rust
- vscode-remote-try-java
- vscode-remote-try-node
- vscode-remote-try-php
- vscode-remote-try-go
- vscode-remote-try-python

these show up in vscode in the try a devcontainer thing https://code.visualstudio.com/docs/devcontainers/containers#_quick-start-try-a-development-container
don't know if this is superseded by the templates thing (can templates do regular files, or just .devcontainer/* files?) or if we want to avoid this for now
![image](https://user-images.githubusercontent.com/61068799/227089670-3b09d726-3c18-43c4-958f-a0fc35163362.png)

https://github.com/orgs/github/repositories?q=codespaces-&type=all&language=&sort=

- codespaces-codeql
- codespaces-blank
- codespaces-rails
- codespaces-react
- codespaces-express
- codespaces-actions-playground
- codespaces-jupyter
- codespaces-django
- codespaces-nextjs
- codespaces-preact
- codespaces-learn-with-me
- codespaces-flask
- codespaces-jetbrains-feedback
- codespaces-getting-started-ml

these show up in https://github.com/codespaces
might be worth it for more involved templates like a full laravel php project with a mysql/mariadb docker-compose devcontainer and all the scaffolding to go with it? that's just an example idea. idk if this is something that we should just say "not devcontainers-community; leave it to the tooling orgs or individuals"
![image](https://user-images.githubusercontent.com/61068799/227090010-cac799d5-554b-4b9c-aec0-0290ee1cad86.png)

i want to stress again **this is an idea** that, given current focus, probably isn't a good fit for _right this instant_ but is something that i'm interested in **possibly** looking into further. particularily because i had an instance over the past two days of fiddling with a docker-compose thing for a mysql php project (hence the example i used above) where i had lackluster and/or outdated demos/examples/docs.

basically i'm saying that i think this "quickstart" template repo thing is an area that doesn't currently have a go-to solution. at least i couldn't find one lol 🤪

**Comment by eitsupi** [comment]
I believe only first-party repositories are integrated into VS Code. (existing since the time of https://github.com/microsoft/vscode-dev-containers)
e.g. microsoft/vscode-dev-containers#1089

**Reply by jcbhmr** [reply]
you're right! it wouldn't be able to be integrated into vscode. 😢 but it's still something that i'd be interested in _exploring_ at sometime in the next few weeks/months. maybe this is something to put under my own jcbhmr/* umbrella instead of trying to stick my own idea under devcontainers-community/*

actually scratch this idea; it looks like somebody else has already tackled the idea of having community templates for popular things: https://github.com/codespaces-examples
